File: go/samples/action_submit_execute_sample.go
package main

import "github.com/rido-min/FluentCards/go/fluentcards"

func createActionSubmitExecuteCard() fluentcards.Card {
	return fluentcards.NewAdaptiveCardBuilder().
		WithVersion("1.4").
		AddTextBlock(func(tb *fluentcards.TextBlockBuilder) {
			tb.WithText("welcome to ac 11").
				WithSize(fluentcards.TextSizeLarge).
				WithWeight(fluentcards.TextWeightBolder)
		}).
		AddTextBlock(func(tb *fluentcards.TextBlockBuilder) {
			tb.WithText("click the buttons below")
		}).
		AddAction(func(a *fluentcards.ActionBuilder) {
			a.Execute("Test AC Action").
				WithData(map[string]any{"message": "button clicked !!"}).
				WithVerb("testAction")
		}).
		AddAction(func(a *fluentcards.ActionBuilder) {
			a.Submit("Open Task Module").
				WithData(map[string]any{"msteams": map[string]any{"type": "task/fetch"}})
		}).
		AddAction(func(a *fluentcards.ActionBuilder) {
			a.Execute("Request File Upload").
				WithVerb("requestFileUpload")
		}).
		Build()
}

File: dotnet/samples/FluentCards.Samples/ActionSubmitExecuteSample.cs
namespace FluentCards.Samples;

/// <summary>
/// Demonstrates creating Submit and Execute actions with custom verbs and data payloads.
/// </summary>
public static class ActionSubmitExecuteSample
{
    /// <summary>
    /// Creates a card with Action.Execute and Action.Submit actions.
    /// </summary>
    /// <returns>An Adaptive Card with Execute and Submit actions.</returns>
    public static AdaptiveCard CreateActionSubmitExecuteCard()
    {
        return AdaptiveCardBuilder.Create()
            .WithVersion("1.4")
            .AddTextBlock(tb => tb
                .WithText("welcome to ac 11")
                .WithSize(TextSize.Large)
                .WithWeight(TextWeight.Bolder))
            .AddTextBlock(tb => tb
                .WithText("click the buttons below"))
            .AddAction(a => a
                .Execute("Test AC Action")
                .WithData("{\"message\":\"button clicked !!\"}")
                .WithVerb("testAction"))
            .AddAction(a => a
                .Submit("Open Task Module")
                .WithData("{\"msteams\":{\"type\":\"task/fetch\"}}"))
            .AddAction(a => a
                .Execute("Request File Upload")
                .WithVerb("requestFileUpload"))
            .Build();
    }
}
